modules/gapi/test/gapi_gpu_test.cpp

```diff
@@ -202,6 +210,243 @@ TEST(GPU, Symm7x7_test)
         EXPECT_EQ(out_mat_gapi.size(), sz);
     }
 }
-#endif
+
+#ifdef HAVE_DIRECTX
+#ifdef HAVE_D3D11
+
+void createD3D11Device(ID3D11Device** device) {
+    IDXGIFactory* adapter_factory = nullptr;
+    HRESULT hr = CreateDXGIFactory(__uuidof(IDXGIFactory),
+                                    reinterpret_cast<void**>(&adapter_factory));
+    if (FAILED(hr)) {
+        CV_Error(cv::Error::BadCallBack, "Cannot create CreateDXGIFactory, error: " +
+                    std::to_string(HRESULT_CODE(hr)));
+    }
+    IDXGIAdapter* intel_adapter = nullptr;
+    UINT adapter_index = 0;
+    const unsigned int refIntelVendorID = 0x8086; // Intel
+    IDXGIAdapter* out_adapter = nullptr;
+
+    while (adapter_factory->EnumAdapters(adapter_index, &out_adapter) != DXGI_ERROR_NOT_FOUND) {
+        DXGI_ADAPTER_DESC adapter_desc{};
+        out_adapter->GetDesc(&adapter_desc);
+        if (adapter_desc.VendorId == refIntelVendorID) {
+            intel_adapter = out_adapter;
+            break;
+        }
+        out_adapter->Release();
+        ++adapter_index;
+    }
+    if (!intel_adapter) {
+        throw SkipTestException("No Intel GPU adapter on aboard.");
+    }
+    intel_adapter->Release();
+
+    *device = nullptr;
+    hr = D3D11CreateDevice(nullptr,
+                           D3D_DRIVER_TYPE_HARDWARE,
+                           nullptr,
+                           0,
+                           nullptr,
+                           0,
+                           D3D11_SDK_VERSION,
+                           &(*device),
+                           nullptr,
+                           nullptr);
+
+    if(FAILED(hr)) {
+        CV_LOG_DEBUG(NULL, "D3D11CreateDevice was failed: " << std::to_string(HRESULT_CODE(hr)));
+        throw SkipTestException("D3D11CreateDevice was failed");
+    }
+    if (*device == nullptr) {
+        CV_LOG_DEBUG(NULL, "OpenCL: D3D11CreateDevice returns empty ID3D11Device\n");
+        throw SkipTestException("D3D11CreateDevice was failed");
+    }
+}
+
+void prepareD3D11Texture2D(const UINT width, const UINT height, std::vector<uint8_t>& test_data,
+                           ID3D11Texture2D** texture,
+                           ID3D11Device** device) {
+    // Create device
+    *device = nullptr;
+    createD3D11Device(device);
+
+    // Create texture description
+    D3D11_TEXTURE2D_DESC desc = {};
+    desc.Width = width;
+    desc.Height = height;
+    desc.MipLevels = 1;
+    desc.ArraySize = 1;
+    desc.Format = DXGI_FORMAT_A8_UNORM; // CV_8U
+    desc.SampleDesc.Count = 1;
+    desc.Usage = D3D11_USAGE_DEFAULT;
+    desc.BindFlags = D3D11_BIND_SHADER_RESOURCE;
+    desc.MiscFlags = D3D11_RESOURCE_MISC_SHARED;
+    desc.CPUAccessFlags = 0; // 0 if CPU access is not required
```

Comment:
you read it  by CPU through cl_mem with EnqueRead by UMat

Could you please find out information how do we need fill this flags in this case?

Reply:
I think, we read from UMat buffer memory. And we copy the UMat buffer from the Texture2D (D3D11Device) (clEnqueueCopyImageToBuffer). Memory of Texture2d is not used for copying to the host, isn't it?

Reply:
Texture as long as cl_mem/cl_image is just a handle to allocated resource in GPU context which can be created with specific access rights. So when you invoke "read" or "write"  operation of "handle" then it use thick layer as runtime + driver+ gpu controller . For our interoperability scenario - i little bit confused because the GPU memory area is the same but it shares TWO handle for that: Texture & cl_mem. So i have no thoughts about on which level ( runtime + driver+ gpu controlle) access rights to resource is actually implemented.

But if openCL allows us to set CL_READ access on Texture  without CPU access during cl_mem creation then (i suppose) we can put it as is. But i suggest to take a look again on openCL funtion with creating cl_mem  and make sure that
1)  we notify proper status in result - i hope CV_Error(status) will print them
2) use access rights flags as not hardcoded CL_READ_WRITE as https://github.com/opencv/opencv/pull/21550/files#diff-a4ca49b5dd017547a48778db3cee476f1732afda1b090369f899e74b8f657799R1569
but passed as  function argument. It is possible to make it default = CL_MEM_READ_WRITE  here
`convertFromD3D11Texture2DtoCLMem(ID3D11Texture2D* pD3D11Texture2D, Device& device, int access_flags)`

Reply:
@mpashchenkov  what do you think?